--- nedit-5.3/source/file.c.orig	Mon Apr 15 12:46:35 2002
+++ nedit-5.3/source/file.c	Thu Jul 18 21:55:58 2002
@@ -1066,7 +1066,7 @@
 */
 void PrintString(const char *string, int length, Widget parent, const char *jobName)
 {
-    char tmpFileName[L_tmpnam];    /* L_tmpnam defined in stdio.h */
+    char *tmpFileName=strdup("/tmp/neditXXXXXX");
     FILE *fp;
     int fd;
 
@@ -1077,17 +1077,13 @@
 	    1. Create a filename
 	    2. Open the file with the O_CREAT|O_EXCL flags
 	So all an attacker can do is a DoS on the print function. */
-    tmpnam(tmpFileName);
+    fd = mkstemp(tmpFileName);
 
     /* open the temporary file */
-#ifdef VMS
-    if ((fp = fopen(tmpFileName, "w", "rfm = stmlf")) == NULL)
-#else
-    if ((fd = open(tmpFileName, O_CREAT|O_EXCL|O_WRONLY, S_IRUSR | S_IWUSR)) < 0 || (fp = fdopen(fd, "w")) == NULL)
-#endif /* VMS */
-    {
+    if ((fp = fdopen(fd, "w")) == NULL) {
     	DialogF(DF_WARN, parent, 1, "Unable to write file for printing:\n%s",
 		"Dismiss", errorString());
+        free(tmpFileName);		
         return;
     }
 
@@ -1097,7 +1093,7 @@
     
     /* write to the file */
 #ifdef IBM_FWRITE_BUG
-    write(fileno(fp), string, length);
+    write(fd, string, length);	
 #else
     fwrite(string, sizeof(char), length, fp);
 #endif
@@ -1106,6 +1102,7 @@
 		jobName, errorString());
 	fclose(fp); /* should call close(fd) in turn! */
     	remove(tmpFileName);
+        free(tmpFileName);		
 	return;
     }
     
@@ -1113,6 +1110,7 @@
     if (fclose(fp) != 0) {
     	DialogF(DF_ERR, parent, 1, "Error closing temp. print file:\n%s",
 		"Dismiss", errorString());
+        free(tmpFileName);		
     	remove(tmpFileName);
 	return;
     }
@@ -1125,6 +1123,7 @@
     PrintFile(parent, tmpFileName, jobName);
     remove(tmpFileName);
 #endif /*VMS*/
+        free(tmpFileName);		
     return;
 }
 
