--- nedit-5.3/makefiles/Makefile.linux~	Wed Aug 28 00:39:26 2002
+++ nedit-5.3/makefiles/Makefile.linux	Wed Aug 28 00:39:26 2002
@@ -16,6 +16,6 @@
 #
 CFLAGS=-O -I/usr/X11R6/include -DUSE_DIRENT -DUSE_LPR_PRINT_CMD
 ARFLAGS=-urs
-LIBS= -L/usr/X11R6/lib -Wl,-Bstatic -lXm -Wl,-Bdynamic -lXp -lXpm -lXext -lXt -lSM -lICE -lX11 -lm
+LIBS=-L/usr/X11R6/lib -lXm -lXp -lXpm -lXext -lXt -lSM -lICE -lX11 -lm
 
 include Makefile.common
